packages/frontend/src/components/DataViz/config/CartesianChartStyling.tsx

```diff
@@ -45,23 +45,22 @@ export const CartesianChartStyling = ({
             return [];
         }
         return currentConfig?.config?.fieldConfig?.y.map((f, index) => {
-            const seriesFormat = Object.values(
+            const foundSeries = Object.values(
                 currentConfig?.config?.display?.series || {},
-            ).find((s) => s.yAxisIndex === index)?.format;
-            const seriesLabel =
-                Object.values(
-                    currentConfig?.config?.display?.series || {},
-                ).find((s) => s.yAxisIndex === index)?.label ??
-                friendlyName(`${f.reference}_${f.aggregation}`);
+            ).find((s) => s.yAxisIndex === index);
 
-            const seriesColor = Object.values(
-                currentConfig?.config?.display?.series || {},
-            ).find((s) => s.yAxisIndex === index)?.color;
+            const seriesFormat = foundSeries?.format;
+            const seriesLabel = foundSeries?.label;
+            const seriesColor = foundSeries?.color;
+            const seriesType = foundSeries?.type;
             return {
                 reference: f.reference,
                 format: seriesFormat,
-                label: seriesLabel,
+                label:
+                    seriesLabel ??
+                    friendlyName(`${f.reference}_${f.aggregation}`),
```

Comment:
f.aggrigation is optional so would it need a fallback if it ends up undefined?

Reply:
Good spot, I believe this could be `undefined` for the semantic viewer, but will be defined for the sql runner. Since this is shared, I'm going to leave it, if that's ok! :)

Reply:
Makes sense, sounds good!